lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -0,0 +1,213 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Imports.
+ */
+/** @jsx jsx */
+import { jsx } from "@emotion/react";
+import * as React from "react";
+import { PureComponent, Fragment } from "react";
+import SignInUpTheme from "../../themes/signInUp";
+import FeatureWrapper from "../../../../../components/featureWrapper";
+import { StyleProvider } from "../../../../../styles/styleContext";
+import { defaultPalette } from "../../../../../styles/styles";
+import { getQueryParams, getRedirectToPathFromURL } from "../../../../../utils";
+import Recipe from "../../../recipe";
+import { getStyles } from "../../../components/themes/styles";
+import { RecipeInterface, LoginAttemptInfo } from "../../../types";
+import { ComponentOverrideContext } from "../../../../../components/componentOverride/componentOverrideContext";
+import { FeatureBaseProps } from "../../../../../types";
+import { formatPhoneNumberIntl } from "react-phone-number-input/min";
+
+type SignInUpState = {
+    error: string | undefined;
+    loaded: boolean;
+    loginAttemptInfo: LoginAttemptInfo | undefined;
+};
+
+type PropType = FeatureBaseProps & {
+    recipe: Recipe;
+};
+
+class SignInUp extends PureComponent<PropType, SignInUpState> {
+    constructor(props: PropType) {
+        super(props);
+
+        let error: string | undefined = undefined;
+        const errorQueryParam = getQueryParams("error");
+        if (errorQueryParam !== null) {
+            if (errorQueryParam === "signin") {
+                error = "Something went wrong. Please try again";
+            } else if (errorQueryParam === "restart") {
+                error = "The link you tried to click expired or was revoked. Please try again.";
+            }
+        }
+        this.state = {
+            loaded: false,
+            loginAttemptInfo: undefined,
```

Comment:
You don't really need `loaded`. You can just check if `loginAttemptInfo !== undefined` instead of checking `if (loaded)`?

Reply:
It's used to make sure we don't pop the email/mobile form in&out before displaying the OTP form when refreshing the page.

Reply:
Let's discuss this over a call.